src/RefractDataStructure.cc

```diff
@@ -703,13 +738,7 @@ namespace drafter {
 
         if (!value.node->valueDefinition.values.empty() && (valuesCount != data.values.size())) {
```

Comment:
Was this `if` loop needed? Won't it be covered by https://github.com/apiaryio/drafter/pull/322/files#diff-337a9124ddc8edd098cabd329b901d4bR703?